cogeo_mosaic/backends/base.py

```diff
@@ -78,10 +76,17 @@ class BaseBackend(BaseReader):
     def __attrs_post_init__(self):
         """Post Init: if not passed in init, try to read from self.input."""
         self.mosaic_def = self.mosaic_def or self._read()
-        self.minzoom = self.mosaic_def.minzoom
-        self.maxzoom = self.mosaic_def.maxzoom
         self.bounds = self.mosaic_def.bounds
 
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if mosaic_tms == self.tms:
+            minzoom, maxzoom = self.mosaic_def.minzoom, self.mosaic_def.maxzoom
+        else:
+            minzoom, maxzoom = self.tms.minzoom, self.tms.maxzoom
+
```

Comment:
if we open the Mosaic with a TMS which is not the mosaic TMS the min/max zoom will default to the TMS (read) zooms IF min/max zoom are not passed by the user

Reply:
a number of comments made on the pr would be better as inline comments in the code